ResourceManager/GitExtensionsControl.cs

```diff
@@ -119,6 +119,28 @@ public virtual bool ProcessHotkey(Keys keyData)
             return hotkey is not null && ExecuteCommand(hotkey.CommandCode);
         }
 
+        /// <summary>
+        /// Runs the associated command, bubbling up the tree if this control does not handle it.
+        /// This mimicks the behavior of <cref>ProcessCmdKey</cref>, in allowing the command request to bubble.
+        /// </summary>
+        public bool DispatchCommand(int command)
+        {
+            for (Control control = this; control != null; control = control.Parent)
+            {
+                if (control is GitExtensionsControl extensionsControl && extensionsControl.ExecuteCommand(command))
+                {
+                    return true;
+                }
+
+                if (control is GitExtensionsFormBase extensionsForm && extensionsForm.ExecuteCommand(command))
+                {
+                    return true;
+                }
+            }
+
+            return false;
+        }
```

Comment:
Sorry for the late finding. But Hotkey commands must not bubble up, only script commands can and need to.
Each control / form has its own enum for hotkeys. They are not globally unique.

Reply:
Hotkey commands already bubble regardless, as enforced by the OS itself. This PR does not change that.

That is not an issue. The contract of `ProcessCmdKey` (and hence `ExecuteCommand`) is to return `true` to stop the bubble as soon as a hotkey (or command) is recognized. So overlap is not an issue.

Besides, the only case where bubbling is done as an `int` (can overlap) and not as a `Keys` (can only overlap for context-dependent hotkeys) is for scripts, which _do_ have a unique non-overlapping integer associated.

Reply:
There is a crucial difference:
```
bool ProcessCmdKey(ref Message msg, Keys keyData)
bool DispatchCommand(int command)
```
`Keys` are generic.
But as long as `int command` identifies a control-specific command, it must not bubble up.
If it is a script, we need to manually bubble it up. This can directly be done in `ExecuteScriptCommand` (inside `GitModuleControl.ExecuteCommand`).

Reply:
> This can directly be done in `ExecuteScriptCommand` (inside `GitModuleControl.ExecuteCommand`).

No, this mustn't be done inside `ExecuteCommand`, because `ExecuteCommand` itself is called multiple times. Your original complaint was valid. This makes the bubbling process `O(n^2)`.

I see two options to resolve this:
1. Move `ExecuteScriptCommand` outside of `ExecuteCommand` and change `DispatchCommand` to `DispatchScriptCommand`, which would call `ExecuteScriptCommand` directly.
2. Change the signature of `ExecuteCommand` to return `void`. Make `ProcessHotkey` always return `true` if a matching hotkey was found. Then the bubbling can occur inside `ExecuteCommand`, as it is guaranteed to be final.

Reply:
Since `n` is negligibly small because there is nearly no nesting of `GitModuleControl`s, I tend to ignore it.

Reply:
What is the suggested change, then? Revert ffb2c60c35b5121feabaa6046a95ccaf5bbc2985? Because you had complaints about that one too. But I can go back, if it's a must. I think it's a huge mistake, but I can do it.

Reply:
Actually, I think the worst-case is if the script has `AskConfirmation` enabled. If you answer `No`, then it will return `false`, causing the script to run again, and hence the popup to appear again, for every control up the tree, so long as the user keeps pressing `No`. I confirmed this happens on `master`, but not on `4.2.1`. Needless to say, this only happens when using a hotkey, and not when running the script.

This PR, in its current form, partially solves this, by reducing it to controls that have an options provider, and the form. But it seems like the correct solution is to force `ExecuteCommand` to be final. i.e. to treat it as if it always returns `true`.

The issue seems to stem from the script engine refactor (ad975756174c381f185d57f72adf690eb339984a), where `RunScript` went from returning `true` if a script exists, to returning `true` if running the script succeeds (and throwing if the script does not exist).

Reply:
Accepted